docs/KFSERVING_DEBUG_GUIDE.md

```diff
@@ -0,0 +1,327 @@
+# Debug KFServing InferenceService Status
+You deployed an InferenceService to KFServing, but it is not in ready state. Go through this step by step guide to understand what failed :heart:.
+```bash
+kubectl get inferenceservices sklearn-iris 
+NAME                  URL   READY   DEFAULT TRAFFIC   CANARY TRAFFIC   AGE
+model-example               False                                      1m
+```
+
+## Check Service Status
+KFServing `InferenceService` creates [Knative Service](https://knative.dev/docs/serving/spec/knative-api-specification-1.0/#service) under the hood to instantiate a 
+serverless container.
+
+If you see `IngressNotConfigured` error, this indicates `Istio Ingress Gateway` probes are failing.
```

Comment:
I would add sub headings like ###IngressNotConfigured so that someone can jump directly to where they are having issues?